### ISSUE TITLE
ML Math

### ISSUE THREAD

**Issue by DustinDoan315** [DustinDoan315/edgeai-learning-roadmap#3]
Dot product and cosine similarity notebook

 Gradients & derivation mini-notebook

 Implement gradient descent from scratch

 Loss functions: MSE + Cross-entropy experiment

 Regularization: L2 + dropout experiment
